workflows/src/main/kotlin/com/r3/developers/apples/workflows/RedeemApplesResponderFlow.kt

```diff
@@ -0,0 +1,30 @@
+package com.r3.developers.apples.workflows
+
+import net.corda.v5.application.flows.CordaInject
+import net.corda.v5.application.flows.InitiatedBy
+import net.corda.v5.application.flows.ResponderFlow
+import net.corda.v5.application.messaging.FlowSession
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.ledger.utxo.UtxoLedgerService
+
+@InitiatedBy(protocol = "redeem-apples")
+class RedeemApplesResponderFlow : ResponderFlow {
+
+    @CordaInject
+    lateinit var utxoLedgerService: UtxoLedgerService
+
+    @Suspendable
+    override fun call(session: FlowSession) {
+        // Receive, verify, validate, sign and record the transaction sent from the initiator
+        utxoLedgerService.receiveFinality(session) { transaction ->
+            /*
+             * [receiveFinality] will automatically verify the transaction and its signatures before signing it.
+             * However, just because a transaction is contractually valid doesn't mean we necessarily want to sign.
+             * What if we don't want to deal with the counterparty in question, or the value is too high,
+             * or we're not happy with the transaction's structure? [UtxoTransactionValidator] (the lambda created
+             * here) allows us to define the additional checks. If any of these conditions are not met,
+             * we will not sign the transaction - even if the transaction and its signatures are contractually valid.
+             */
+        }
```

Comment:
Only comments in the block ... shouldn't we have code ?

Reply:
I'm following the prescribed steps in [Building Your First CordApp > Write Flows > Check Your Work](https://docs.r3.com/en/platform/corda/5.0/developing-applications/basic-cordapp/flows.html#finalize-the-transaction-in-createandissueapplestampresponderflow) and it suggests to leave the flow blank for now.

Unless you think we should write the flow?